spec/predicates/human-review-vcs.md

```diff
@@ -0,0 +1,155 @@
+# Predicate type: VCS Code Reviews
+
+Type URI: (tentative) https://in-toto.io/attestation/human-review/vcs/v0.1
+
+Version: 0.1.0
+
+## Purpose
+
+This attestation type is used to describe code review approvals of the simplest
+kind. The general expectation is that reviews are tied to specific changes in a
+version control system (VCS) such as Git.
+
+## Use Cases
+
+Changes to source code are reviewed by other members of the development team.
+The review may be performed using systems like Gerrit or via reviewing
+mechanisms built into version control forges such as GitHub and GitLab.
+
+### Threshold of Reviews on Pull Requests
+
+GitHub allows repository owners to create protection rules that require a
+threshold of approvals before a pull request can be merged. This predicate can
+be used to express meeting such a policy. In this case, GitHub would issue a
+signed attestation identifying all the reviewers for a pull request. The
```

Comment:
The idea is that this sort of attestation wouldn't be generated until the threshold is met, right?
```suggestion
signed attestation identifying all accepting reviewers for a pull request. The
```

Reply:
Also, this may be getting too much into the weeds.... the current GitHub rule for requiring approvals before a PR can be merged reads "When enabled, pull requests targeting a matching branch require a number of approvals _and no changes requested_ before they can be merged" (emphasis mine). 

I've personally experienced this: even though I had the threshold number of needed approvals, a third reviewer had requested changes in the past and not re-reviewed the PR. It wasn't until their review was either dismissed or updated that the PR was able to be merged. This may be a GH-specific rule, but what is the expectation for generating this attestation in this sort of case? The current text only describes how the attestation subject is chosen wrt a PR merge rule, but it may be worth clarifying the exact point or operation upon which this attestation should be generated (e.g., after the approved PR has been merged).

Reply:
These are great questions and have (somewhat) come up in the SLSA Source track doc as well. It all comes down to when the attestation is created and signed + what it's associated with. I authored this take on the VCS review predicate thinking it'd be generated at merge time. But this obfuscates cases where stale approvals aren't dismissed (which is I think the more common configuration). @zachariahcox has made this point in the Source track doc if I'm not mistaken, and his take on a similar predicate identifies associates each individual review of the changeset with the commit at the tip of the topic branch. Settling when the predicate is generated will help us decide what and how its contents must be presented IMO.